src/hotspot/cpu/x86/macroAssembler_x86.cpp

```diff
@@ -2989,91 +2989,45 @@ void MacroAssembler::push_IU_state() {
 void MacroAssembler::push_cont_fastpath() {
   if (!Continuations::enabled()) return;
 
-#ifndef _LP64
-  Register rthread = rax;
-  Register rrealsp = rbx;
-  push(rthread);
-  push(rrealsp);
-
-  get_thread(rthread);
-
-  // The code below wants the original RSP.
-  // Move it back after the pushes above.
-  movptr(rrealsp, rsp);
-  addptr(rrealsp, 2*wordSize);
-#else
   Register rthread = r15_thread;
   Register rrealsp = rsp;
-#endif
 
   Label done;
   cmpptr(rrealsp, Address(rthread, JavaThread::cont_fastpath_offset()));
```

Comment:
While you're here, can you change rthread to r15_thread?

Reply:
Yeah, I don't mind, changed.